dart/constraint/ContactConstraint.cpp

```diff
@@ -850,8 +910,10 @@ ContactConstraint::getTangentBasisMatrixODE(const Eigen::Vector3d& n)
   // Note: a possible speedup is in place for mNumDir % 2 = 0
   // Each basis and its opposite belong in the matrix, so we iterate half as
   // many times
-  T.col(0) = tangent;
-  T.col(1) = Eigen::Quaterniond(Eigen::AngleAxisd(0.5_pi, n)) * tangent;
+  // The first column is the same as mFirstFrictionalDirection unless
+  // mFirstFrictionalDirection is parallel to the normal
+  T.col(0) = Eigen::Quaterniond(Eigen::AngleAxisd(0.5_pi, n)) * tangent;
+  T.col(1) = tangent;
```

Comment:
It's not the part of this change, but I would like to negate the direction of `tangent` so that `T.col(0).cross(T.col(1)) == n` (currently, it's `T.col(1).cross(T.col(0)) == n`) because I believe it'd be more intuitive. This change wouldn't affect the behavior (but just flipping the sign of impulse for the second tangential friction).

It can be done by changing the order of all the cross products in this function. For example, [line 813](https://github.com/dartsim/dart/pull/1427/files#diff-ade956ee6a4569052e9d25915f8715aaL813) should be changed to:
```cpp
Eigen::Vector3d tangent = n.cross(mFirstFrictionalDirection);
```
or simply negating the final `tangent`. I'd prefer changing the order of the cross products though.

Reply:
I'm not opposed to that, but I'll bring @azeey into the conversation since he made this specific change

Reply:
Negating the direction of `tangent` sounds fine to me if it doesn't affect the behavior. I think the main change here is that `T.col(0)` has to be parallel to the first frictional direction.

Reply:
reversed in 5fe40b747b243875b655663dc458edd7ef3f8a1c